File: src/main/java/com/gusrylmubarok/reddit/backend/repository/CommentRepository.java
package com.gusrylmubarok.reddit.backend.repository;

import com.gusrylmubarok.reddit.backend.model.Comment;
import com.gusrylmubarok.reddit.backend.model.Post;
import com.gusrylmubarok.reddit.backend.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CommentRepository extends JpaRepository<Comment, Long> {
    List<Comment> findByPost(Post post);
    List<Comment> findAllByPost(Post post);
    List<Comment> findAllByUser(User user);
}

File: src/main/java/com/reddot/repository/CommentRepository.kt
package com.reddot.repository

import com.reddot.entity.Comment
import com.reddot.entity.Post
import com.reddot.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface CommentRepository : JpaRepository<Comment, Long> {
    fun findByPost(post: Post): List<Comment>
    fun findAllByUser(user: User): List<Comment>
}